godot-macros/src/class/godot_api.rs

```diff
@@ -67,21 +67,32 @@ impl BoundAttr {
     }
 }
 
+/// Holds information known from a signal's definition
+struct SignalDefinition {
+    signature: Function,
+    attributes: Vec<Attribute>,
+}
+
 /// Codegen for `#[godot_api] impl MyType`
 fn transform_inherent_impl(mut decl: Impl) -> Result<TokenStream, Error> {
     let class_name = util::validate_impl(&decl, None, "godot_api")?;
     let class_name_obj = util::class_name_obj(&class_name);
     let (funcs, signals) = process_godot_fns(&mut decl)?;
 
+    let mut signal_cfg_attrs: Vec<Vec<&Attribute>> = Vec::new();
     let mut signal_name_strs: Vec<String> = Vec::new();
     let mut signal_parameters_count: Vec<usize> = Vec::new();
     let mut signal_parameters: Vec<TokenStream> = Vec::new();
 
-    for signature in signals {
+    for signal in &signals {
+        let SignalDefinition {
+            signature,
+            attributes,
+        } = signal;
         let mut param_types: Vec<TyExpr> = Vec::new();
         let mut param_names: Vec<String> = Vec::new();
 
-        for param in signature.params.inner {
+        for param in &signature.params.inner {
```

Comment:
Just to add here that the usage of borrowing here was mostly to satisfy the borrow checker, as `extract_cfg_attrs` returns a reference to the given attributes, so they shouldn't die with `signal` in this loop if we want to use the cfg attrs later. We could, of course, just make `util::extract_cfg_attrs` return `Vec<Attribute>` by cloning them (that just didn't seem necessary, in principle). Feel free to decide the best approach here.

Reply:
Okay! Changed it.